pkg/operator/certrotationcontroller/certrotationcontroller.go

```diff
@@ -496,4 +496,6 @@ func (c *CertRotationController) Run(ctx context.Context, workers int) {
 	for _, certRotator := range c.certRotators {
 		go certRotator.Run(ctx, workers)
 	}
+
+	<-ctx.Done()
```

Comment:
have you audited all consumers that they cope with a blocking `Run` func?

Reply:
yes